tests/unit/Middleware/RecordTransactionTest.php

```diff
@@ -0,0 +1,147 @@
+     */
+    private $request;
+
+    /**
+     * @var Illuminate\Http\Response
+     */
+    private $response;
+
+    private $middleware;
+
+    /**
+     * @var PhilKra\Events\Transaction
+     */
+    private $transaction;
+
+    protected function _before()
+    {
+        $this->agent = Mockery::mock(Agent::class);
+        $this->transaction = Mockery::mock(Transaction::class)->makePartial();
+        $this->request = Request::create('/ping', 'GET');
+        $this->response = Mockery::mock(Response::class)->makePartial();
+        $this->response->headers = new ResponseHeaderBag();
+
+        Config::shouldReceive('get')
+            ->with('elastic-apm-laravel.transactions.useRouteUri')
+            ->andReturn(false);
+
+        $this->middleware = new RecordTransaction($this->agent);
+    }
+
+    public function testStartTransaction()
+    {
+        $this->agent->shouldReceive('startTransaction')
+            ->once()
+            ->withArgs(function ($transaction_name, $context, $request_time) {
+                $this->assertEquals('GET /ping', $transaction_name);
+                $this->assertEquals([], $context);
+                $this->assertNotNull($request_time);
+
+                return true;
+            })
+            ->andReturn($this->transaction);
+
+        $this->middleware->handle($this->request, function () {
+            return $this->response;
+        });
+    }
+
+    public function testTransactionMetadata()
+    {
+        $this->agent->shouldReceive('startTransaction')
+            ->once()
+            ->andReturn($this->transaction);
+
+        $this->response->shouldReceive('getStatusCode')
+            ->times(2)
+            ->andReturn(200);
+
+        $this->middleware->handle($this->request, function () {
+            return $this->response;
+        });
+
+        $this->assertEquals(200, $this->transaction->getMetaResult());
+        $this->assertEquals('HTTP', $this->transaction->getMetaType());
+    }
+
+    public function testTransactionContext()
+    {
+        $this->agent->shouldReceive('startTransaction')
+            ->once()
+            ->andReturn($this->transaction);
+
+        $this->response->shouldReceive('getStatusCode')
+            ->times(2)
+            ->andReturn(200);
+
+        $this->middleware->handle($this->request, function () {
+            return $this->response;
+        });
+
+        $context = $this->transaction->getContext();
+
+        Assert::assertArraySubset([
+            'finished' => true,
+            'headers_sent' => true,
+            'status_code' => 200,
+            'headers' => [],
+        ], $context['request']['response']);
+
+        $this->assertEquals([
+            'id' => null,
+            'ip' => '127.0.0.1',
+            'user-agent' => 'Symfony/3.X',
+        ], $context['user']);
+    }
+
+    public function testTransactionTerminate()
+    {
+        // Don't expect Config::get to be called
+        Config::mockery_teardown();
```

Comment:
wow, is this a work-around for changing mockery expectation? I'm not sure I'd not prefer to just copy-paste the expectation from `_before` to all methods that need it.